packages/caliper-fabric/test/connector-configuration/ConnectionProfileDefinition.js

```diff
@@ -436,4 +436,460 @@ describe('A Connection Profile Definition', async () => {
+                'No peers property can be found in the connection profile provided'
+            );
+        });
+
+        it('should throw an error if pem file provided through tlsCAcets.pem is not valid', async () => {
+            const invalidPemConnectionProfile = {
+                peers: {
+                    'peer0.org1.example.com': {
+                        tlsCACerts: {
+                            pem: 'MIICWDCCAf6gAwIBAgIRAMpSgWFjDHOohXa0R6e9THgwCgYIKoZIzj0EAwIwdjEL\nMAkGA1UEBhMCVVMxEzARBgNVBAgTCkNhbGlmb3JuaWExFjAUBgNVBAcTDVNhbiBG\ncmFuY2lzY28xGTAXBgNVBAoTEG9yZzEuZXhhbXBsZS5jb20xHzAdBgNVBAMTFnRs\nc2NhLm9yZzEuZXhhbXBsZS5jb20wHhcNMjAwOTA3MTE0MjAwWhcNMzAwOTA1MTE0\nMjAwWjB2MQswCQYDVQQGEwJVUzETMBEGA1UECBMKQ2FsaWZvcm5pYTEWMBQGA1UE\nBxMNU2FuIEZyYW5jaXNjbzEZMBcGA1UEChMQb3JnMS5leGFtcGxlLmNvbTEfMB0G\nA1UEAxMWdGxzY2Eub3JnMS5leGFtcGxlLmNvbTBZMBMGByqGSM49AgEGCCqGSM49\nAwEHA0IABMdLvSUDIjWYuBw4YVvJEW6ifFLymOAX7GKY6btVPDlkdeJ8vZErXLMz\nJWjivr/L5V2YnZqv0OWPMMfPv+zH+RGjbTBrMA4GA1UdDwEB/wQEAwIBpjAdBgNV\nHSUEFjAUBggrBgEFBQcDAgYIKwYBBQUHAwEwDwYDVR0TAQH/BAUwAwEB/zApBgNV\nHQ4EIgQg5fOhyzwaLKm3t54/H4b0aTe7/nGPyJZNh9IFRK6fDaAwCgYIKoZIzj0E\nAwIDSAAwRQIhAKEny//JY7GXZ/THsQIvUTYmXsjP/bLTI/VuLX7TzcefAiBYoSyY\ny90rdprI6Mp6RPiqjVf02P5ZC86UkP0Vw4pfiQ==\n-----END CERTIFICATE-----\n'
+                        },
+                    }
+                }
+            };
+            const connectionProfileDefinition = new ConnectionProfileDefinition(mspId, {
+                loadedConnectionProfile: invalidPemConnectionProfile,
+                discover: false
+            });
+            await connectionProfileDefinition.getTlsCertForPeer(peer).should.be.rejectedWith(
+                `pem provided for ${peer} in the connection profile .tlsCACerts.pem is not valid`
+            );
+        });
+        it('should throw an error if path provided through tlsCAcets.path does not point to a valid cert', async () => {
+            const invalidPemConnectionProfile = {
+                peers: {
+                    'peer0.org1.example.com': {
+                        tlsCACerts: {
+                            path:'.test/sample-configs/invalid.pem'                        },
+                    }
+                }
+            };
+            const path = CaliperUtils.resolvePath(invalidPemConnectionProfile.peers[peer].tlsCACerts.path);
+            const connectionProfileDefinition = new ConnectionProfileDefinition(mspId, {
+                loadedConnectionProfile: invalidPemConnectionProfile,
+                discover: false
+            });
+            await connectionProfileDefinition.getTlsCertForPeer(peer).should.be.rejectedWith(
+                `path property does not point to a file that exists at ${path} for ${peer}`
+            );
+        });
+    });
+
+
+    describe('when getting the grpc endpoint for a peer', () => {
+
+        const peer = 'peer0.org1.example.com';
+
+        it('should return the parsed for a grpcs url', async () => {
```

Comment:
this description is not quite right, but would suggest `should return the correct endpoint for a grpcs url`